src/coldcard.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Provides classes for interacting with a Coldcard via TXT/JSON/PSBT files
+ *
+ * The following API classes are implemented:
+ *
+ * * ColdcardExportExtendedPublicKey
+ * * ColdcardSignMultisigTransaction
+ * * ColdcardMultisigWalletConfig
+ *
+ * @module coldcard
+ */
+import {
+  IndirectKeystoreInteraction,
+} from "./interaction";
+
+export const COLDCARD = 'coldcard';
+export const WALLET_CONFIG_VERSION = "0.0.1";
+
+
+/**
+ * Base class for indirect interactions with Coldcard.
+ *
+ * @extends {module:interaction.IndirectKeystoreInteraction}
+ */
+export class ColdcardInteraction extends IndirectKeystoreInteraction {
+
+}
+
+/**
+ * Base class for coldcard interactions (reading/writing files)
+ *
+ * @extends {module:hermit.HermitInteraction}
+ */
+export class ColdcardFileReader extends ColdcardInteraction {
+
+}
+
+export class ColdcardExportExtendedPublicKey extends ColdcardFileReader {
+
+}
+
+/**
+ * Returns signature request data to display via PSBT for Coldcard and returns the PSBT
+ * data back to the calling application
+ */
+export class ColdcardSignTransaction extends ColdcardFileReader {
+
+}
+
+/**
+ * Returns a valid multisig wallet config text file to send over to a Coldcard
+ *
+ * NOTE: only the root xfp of the signing device is required to be correct, but we
+ * recommend only setting up the multisig wallet on the Coldcard with complete xfp
+ * information. We define isSupported() to check exactly this condition.
+ *
+ * This is an example Coldcard config file from
+ * https://coldcardwallet.com/docs/multisig
+ *
+ * # Coldcard Multisig setup file (exported from 4369050F)
+ * #
+ * Name: MeMyself
+ * Policy: 2 of 4
+ * Derivation: m/45'
+ * Format: P2WSH
+ *
+ * D0CFA66B: tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9
+ * 8E697B74: tpubD97nVL37v5tWyMf9ofh5rzn...XgSc
+ * BE26B07B: tpubD9ArfXowvGHnuECKdGXVKDM...FxPa
+ * 4369050F: tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n
+ *
+ */
+export class ColdcardMultisigWalletConfig {
```

Comment:
Something that feels like it would be nice to have are methods that are named like `fromColdcardConfig` and `toColdcardConfig`. The `from` would be a tool to parse what is retrieved from the config file and turn it into a class that has some actual functionality. The `to` method is basically what I'm gathering `adapt` does. 

But maybe this is more of an argument for what's been discussed elsewhere about having a general purpose `MultisigWalletConfig` class which in addition to the above methods could be a lot more useful and flexible and include things like `toElectrumConfig` (and `from`), `toCaravanConfig`, etc. Doing it this way has a nice added benefit of once you've imported/converted, uc-wallets/bitcoin is then providing a uniform interface for interacting with a multisig wallet config that can be passed around and converted as necessary. 

Starting with what's here as a base and just generalizing the naming, even if it only works for coldcard for now shouldn't involve too much while offering more flexibility for the long term.

Reply:
This sounds good on the surface but I don't how it would work in practice given the missing information (bip32 paths), for example, in the current Coldcard config. The `fromColdcardConfig` e.g. does not give enough information for us to create a CaravanConfig or an ElectrumConfig.

Reply:
But then this can be handled in the conversion. If you import a config via `fromColdcardConfig`, we could then have methods to add missing information, which the consumer of this library could decide any number of ways they want to retrieve it. Then if a `.toElectrumConfig` is called if it's not available on the current object, you would get an error (and a very specific error telling you why an electrum config isn't possible to generate.

Reply:
I actually had to do something similar on another project when dealing with LSATs (lightning service authentication tokens). Basically there were a lot of different ways to consume and generate lsats. They are typically, like the Coldcard Config from what I can tell, sent in http headers as text encoded in a particular way. But I needed a way so that specific information could be consumed and easily read. So, this tool would read the plain text, learn where things were broken up and then add as properties on the object. If you wanted to convert back as an LSAT or as a challenge, the conversion method determines if it has enough information and throws a descriptive error if it doesn't. 

Here's the main file I'm thinking about: https://github.com/Tierion/lsat-js/blob/master/src/lsat.ts
You can see in [fromChallenge](https://github.com/Tierion/lsat-js/blob/master/src/lsat.ts#L347) it will look for all the text elements it expects.
Meanwhile [toChallenge](https://github.com/Tierion/lsat-js/blob/master/src/lsat.ts#L245) throws if it doesn't have an invoice property attached to it.

Anyway, sorry to bring a totally random and unrelated project into this. It's just something I was working on relatively recently and felt relevant. In short, I think the ability for this Class to actually have the ability to tell us it can't be converted to an ElectrumConfig is super powerful as a tool and feels like it belongs together with the other conversion classes since the config, unlike the keystore classes, are kind of supposed to be interoperable in my opinion.

Reply:
Yeah - I hear that. I'm open to more discussion 1on1 or with a wider audience, but I'd argue that if you want an Electrum config from a Coldcard, there's a path on the device for that (Multisig Wallet Details > Electrum Wallet) and it'll output the file you want to the SD card.

Reply:
What about for someone not using a coldcard though? What if I want to go from Electrum to Caravan? It also could be useful in other UI settings.

Reply:
What if you have a UI like in caravan that lets you compose a config that you want to be able to import into either Coldcard or Electrum? You could have the form just pass the object to this class and depending on what is selected, spit out the correctly formatted text/file for Caravan, Coldcard, Electrum, or any other format that ends up getting created which we could extend this base class to support.